lib/ea_restaurant_data_loader_web/controllers/oauth2_controller.ex

```diff
@@ -21,4 +21,21 @@ defmodule EaRestaurantDataLoaderWeb.Controllers.Oauth2Controller do
 
     send_resp(conn, :ok, login_json_response)
   end
+
+  def refresh_token(conn, _params) do
+    access_token = conn.body_params["access_token"]
+    refresh_token = conn.body_params["refresh_token"]
+    client_id = conn.body_params["client_id"]
+    client_secret = conn.body_params["client_secret"]
+
+    {_, refresh_token_json_response} =
+      try do
+        Oauth2Service.refresh_token(refresh_token, access_token, client_id, client_secret)
+        |> ApplicationUtil.parse_to_json()
+      rescue
+        _ in MatchError -> raise TokenExpiredError
```

Comment:
Not all errors are meant to be `TokenExpiredError` I don't think a `Matcherror` should be a `TokenExpiredError`

Reply:
Done. i refactored the refresh token service to raise expired token intead of matching with {:ok,_}